content/intro-to-storybook/ember/en/data.md

```diff
@@ -71,189 +82,87 @@ export const reducer = (state, action) => {
-];
-
-export const Default = () => ({
-  template: hbs`<div style="padding: 3rem"><PureTaskList @tasks={{this.tasks}} @pinTask={{fn this.onPinTask}} @archiveTask={{fn this.onArchiveTask}}/></div>`,
-  context: {
-    tasks: defaultTasksData,
-    ...actionsData,
-  },
-});
-
-export const withPinnedTasks = () => ({
-  template: hbs`<div style="padding: 3rem"><PureTaskList @tasks={{this.tasks}} @pinTask={{fn this.onPinTask}} @archiveTask={{fn this.onArchiveTask}}/></div>`,
-  context: {
-    tasks: withPinnedTasksData,
-    ...actionsData,
-  },
-});
-export const Loading = () => ({
-  template: hbs`<div style="padding: 3rem"><PureTaskList @tasks={{this.tasks}} @loading={{true}}/></div>`,
-  context: {
-    tasks: [],
-  },
-});
-export const Empty = () => ({
-  template: hbs`<div style="padding: 3rem"><PureTaskList @tasks={{this.tasks}}/></div>`,
-  context: {
-    tasks: [],
-  },
-});
+And we use our earlier created presentational `<TaskList>` component, mount it
+at the route template and connect it with the actions in the controller and the
+data from the route.
+
+```hbs
+{{!--app/tasks/template.hbs --}}
+<TaskList
+  @tasks={{@model}}
+  @pinTask={{this.pinTask}}
+  @archiveTask={{this.archiveTask}}
+/>
 ```
 
-<video autoPlay muted playsInline loop>
-  <source
-    src="/intro-to-storybook/finished-tasklist-states.mp4"
-    type="video/mp4"
-  />
-</video>
-
-Similarly, we need to use `PureTaskList` in our Qunit test:
-
-```javascript
-// tests/integration/task-list-test.js
-import { module, test } from 'qunit';
-import { setupRenderingTest } from 'ember-qunit';
-import { render } from '@ember/test-helpers';
-import { hbs } from 'ember-cli-htmlbars';
-module('Integration | Component | TaskList', function(hooks) {
-  setupRenderingTest(hooks);
-  const taskData = {
-    id: '1',
-    title: 'Test Task',
-    state: 'TASK_INBOX',
-    updatedAt: new Date(2018, 0, 1, 9, 0),
-  };
-  const tasklist = [
-    { ...taskData, id: '1', title: 'Task 1' },
-    { ...taskData, id: '2', title: 'Task 2' },
-    { ...taskData, id: '3', title: 'Task 3' },
-    { ...taskData, id: '4', title: 'Task 4' },
-    { ...taskData, id: '5', title: 'Task 5' },
-    { ...taskData, id: '6', title: 'Task 6 (pinned)', state: 'TASK_PINNED' },
-  ];
-
-  test('renders pinned tasks at the start of the list', async function(assert) {
-    this.tasks = tasklist;
-    await render(hbs`<PureTaskList @tasks={{this.tasks}}/>`);
-    assert.dom('[data-test-task]:nth-of-type(1)').hasClass('TASK_PINNED');
-  });
-});
-```
+We will keep the presentational `<TaskList>` decoupled from the data-loading
+mechanism to enforce separation of concerns and a good architecture overall.
+However, also our current implementation of using a route and a controller
+doesn't feel overall pleasing. We can use a container component instead, that
+is also responsible for data loading and visually managing its state. We will do
+this in the next step with the introduction of screen components.
```

Comment:
@gossi we can probably phrase this a bit different. Probably go with something like:

"With this we've accomplished what we've set out to do, we've managed to setup a data persistance layer and also we've managed to keep the components decoupled, With this enforcing some best practices. But this type of implementation doesn't feel overall pleasing. We could use a container component instead, which will be responsible for loading the data and also visually managing its state. And that's what we'll do in the next chapter when we introduce screen components". Something like this?